### PR TITLE
adding ParticipationUpdates and CompactCertState to BlockHeader

### PR DESCRIPTION
**Pull request by barnjamin**
Addresses #305

### CODE REVIEW

**Comment by barnjamin** [comment]
This should probably have its own test that somehow also fails if go-algorand changes?

Can we just use the structs defined in go-algorand for this?

**Comment by pbennett** [comment]
Thanks for this fix - this has happened before as well - the duplication of these core structures either needs unified, or there needs to be good tests to catch missing additions.
This caused a halt on my block watcher on testnet which has me concerned over what other sdks might have the same issue and the impact on mainnet for integrators.

File: types/block.go [hunk 1]
```diff
@@ -152,6 +172,29 @@ type (
 		NextProtocolSwitchOn   Round  `codec:"nextswitch"`
 	}
 
+	// CompactCertState tracks the state of compact certificates.
+	CompactCertState struct {
```
[the remark this answers is not in the document]
Comment (barnjamin) [reply types/block.go]: fair enough, merged Wills suggestion

File: types/block.go [hunk 2]
```diff
@@ -92,6 +92,26 @@ type (
 		// transactions have ever been committed (since TxnCounter
 		// started being supported).
 		TxnCounter uint64 `codec:"tc"`
+
+		// CompactCert tracks the state of compact certs, potentially
+		// for multiple types of certs.
+		//msgp:sort protocol.CompactCertType protocol.SortCompactCertType
+		//CompactCert map[protocol.CompactCertType]CompactCertState `codec:"cc"`
```
Comment (algoidan) [inline types/block.go:99]: nits: should we remove this comment as well?

Comment (barnjamin) [reply types/block.go:99]: Yep, I'll push this change as well

**Review by algoidan — APPROVED** [review]

**Review by winder — APPROVED** [review]

**Pull request merged**